Update display of switching current network

amerkadicE opened the pull request:
Fixes: #12691

Explanation:  

Design is updated when switching to a new network.

Manual testing steps:  
  - Use  https://chainlist.org/  to switch network in MetaMask
  - New design for network switching should be shown
  - 
![image](https://user-images.githubusercontent.com/97883527/151781554-6de4e0c9-c8c3-44f0-9e93-0ebe9b61f479.png)

Commit:
Update display of switching current network

Fix lint error

fix: top margin fix to remove scroll

fix resolve github comments

Fix lint

Fix image

Fix arrow for rtl direction

svg compress

amerkadicE commented:
I have read the CLA Document and I hereby sign the CLA

darkwing commented:
This needs a rebase but looks *awesome*!  Well done!

jpuri commented on ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js:
```diff
@@ -0,0 +1,104 @@
+        </Box>
+      </Box>
+    </Box>
+  );
```
May be level of nesting `Box` component above can be reduced.

amerkadicE replied:
Done, absolute position and one Box component removed.

jpuri commented on ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js:
```diff
@@ -0,0 +1,104 @@
+          strokeLinejoin="round"
+          strokeDasharray="8 7"
+        />
+      </svg>
```
Across the app we have used `svg` with `img` tag, may be we should do same here.

amerkadicE replied:
I added a new svg file for the broken line and used it on two places in the project.

jpuri commented on app/images/arrow-right-blue.svg:
```diff
@@ -0,0 +1,19 @@
+<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
```
SVG can be minified using https://jakearchibald.github.io/svgomg/

amerkadicE replied:
Done.

brad-decker commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  }
+
+  &__icon {
+    display: block;
```
display block can be achieved on the Box component

amerkadicE replied:
Done.

brad-decker commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  &__center-icon {
+    @include H7;
+
+    display: flex;
+    position: relative;
+    justify-content: center;
+    align-items: center;
```
3 of 4 of these styles can be achieved with the props on Box component

amerkadicE replied:
Done.

brad-decker commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  }
+
+  &__center-icon {
+    @include H7;
```
What text does this apply to?

amerkadicE replied:
I was using this at first, but gave up on that solution. I forgot to remove it from scss, thank you for noticing. It is removed now.

brad-decker commented on ui/pages/confirmation/confirmation.js:
```diff
@@ -185,23 +184,20 @@ export default function ConfirmationPage() {
+          <SiteIcon
+            icon={originMetadata.iconUrl}
+            name={originMetadata.hostname}
+            size={36}
           />
-        </Box>
-        <Box justifyContent="center" padding={[1, 4, 4]}>
           <Chip
             label={stripHttpsScheme(originMetadata.origin)}
-            leftIcon={
-              <SiteIcon
-                icon={originMetadata.iconUrl}
-                name={originMetadata.hostname}
-                size={32}
-              />
-            }
+            borderColor={COLORS.WHITE}
```
do not change these styles just yet, we may move to a place where no confirmations need these but until then if you rebase on develop you should see the chip has been removed and a template prop added to control the display of the network display component.

amerkadicE replied:
I resolve this conflicts as agreed in this slack conversation. 
https://consensys.slack.com/archives/C02GH833686/p1643894366164449

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+.confirmation-network-switch {
+  display: flex;
+  height: 100%;
+  justify-content: center;
```
Done.

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  &__result {
+    @include H3;
+
+    position: absolute;
```
Absolute position removed.

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  justify-content: center;
+
+  &__result {
+    @include H3;
```
I was using this at first, but gave up on that solution. I forgot to remove it from scss, thank you for noticing. It is removed now.

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+    justify-content: space-between;
+    align-items: center;
+    text-align: center;
+    color: var(--Black-100);
```
I was using this at first, but gave up on that solution. I forgot to remove it from scss, thank you for noticing. It is removed now.

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+  }
+
+  &__icons {
+    display: flex;
```
Done.

mirjanaKukic commented:
Verified by QA

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/index.scss:
```diff
@@ -0,0 +1,56 @@
+      width: 64px;
+      position: absolute;
+      display: flex;
+      justify-content: center;
```
Done

amerkadicE commented:
Hi @brad-decker,
Can you take a look into this PR?
Is it alright now, can we move forward with the PR?

darkwing reviewed (dismissed):
Wow @amerkadicE , this is amazing!  I love this!  One very small detail I noticed.  When I switch to RTL direction, the arrow stay the same direction, while the icons/networks change sides, which makes the graphic look a bit misleading:

<img width="1295" alt="RTL" src="https://user-images.githubusercontent.com/46655/157062972-aa73c763-3ac1-427b-b84c-c4ab95655f0d.png">

For `.confirmation-network-switch__check`, we could do:

```
[dir='rtl'] .confirmation-network-switch__check {
  transform: rotate(180deg);
}
```

amerkadicE commented:
> Wow @amerkadicE , this is amazing! I love this! One very small detail I noticed. When I switch to RTL direction, the arrow stay the same direction, while the icons/networks change sides, which makes the graphic look a bit misleading:
> 
> <img alt="RTL" width="1295" src="https://user-images.githubusercontent.com/46655/157062972-aa73c763-3ac1-427b-b84c-c4ab95655f0d.png">
> 
> For `.confirmation-network-switch__check`, we could do:
> 
> ```
> [dir='rtl'] .confirmation-network-switch__check {
>   transform: rotate(180deg);
> }
> ```

Thank you @darkwing. 
I updated arrow direction for RTL HTML.

adonesky1 commented on app/images/arrow-right-blue.svg:
```diff
@@ -0,0 +1,19 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" fill="none">
```
I think we could compress this SVG using something like this https://vecta.io/nano> ?

amerkadicE replied:
Done

adonesky1 commented on ui/pages/pages.scss:
```diff
@@ -2,6 +2,7 @@
 @import 'add-collectible/index';
 @import 'import-token/index';
 @import 'asset/asset';
+@import 'confirmation/components/confirmation-network-switch/index';
```
I think it probably makes more sense to import this file in `confirmation/confirmation.scss` which is then imported below on line 12

amerkadicE replied:
Done

adonesky1 commented:
Just a few small things (comments above) but otherwise LGTM

danjm reviewed (dismissed):
LGTM

brad-decker commented on ui/pages/confirmation/templates/switch-ethereum-chain.js:
```diff
@@ -46,9 +47,10 @@ function getValues(pendingApproval, t, actions) {
         children: t('switchEthereumChainConfirmationDescription'),
         props: {
           variant: TYPOGRAPHY.H7,
+          color: COLORS.UI4,
```
I believe this should be updated to use our new colors from the design system so that dark mode and other themes in the future will work appropriately. @georgewrmarshall could you advise on the best way to future proof this new work?

georgewrmarshall reviewed (changes requested):
Looking good! We recently updated all of our colors to improve consistency and enable themability for the extension. I'm still working on improving the documentation around color and styles in the mean time I've created a PR with some suggested updates https://github.com/MetaMask/metamask-extension/pull/14255

Commit:
Enable dark mode for network switch

amerkadicE commented on app/images/arrow-right-blue.svg:
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="40" height="40" fill="none" xmlns:v="https://vecta.io/nano"><g filter="url(#A)"><path fill="#61ba00" stroke="#61ba00" d="M34.5 19c0 8.008-6.492 14.5-14.5 14.5S5.5 27.008 5.5 19 11.992 4.5 20 4.5 34.5 10.992 34.5 19Z"/><path stroke="#fff" stroke-width="2" d="m13.3 19.72 3.513 3.6 9.387-9.3"/><circle cx="20" cy="19" r="15" fill="#037dd6"/><path fill="# ...
```
Removed

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js:
```diff
@@ -0,0 +1,118 @@
+            size={64}
+          />
+          <Typography
+            color={COLORS.BLACK}
```
Done

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js:
```diff
@@ -0,0 +1,118 @@
+            size={64}
+          />
+          <Typography
+            color={COLORS.BLACK}
```
Done

amerkadicE commented on ui/pages/confirmation/components/confirmation-network-switch/confirmation-network-switch.js:
```diff
@@ -0,0 +1,118 @@
+    chainId: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+  }),
+};
```
Done

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/97883527/160839755-c896b419-5018-4463-8f12-0efa072053af.png">

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/97883527/160839801-db0699fd-6d78-4e6e-ab20-5409a4406caa.png">

amerkadicE replied:
Done

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/97883527/160839755-c896b419-5018-4463-8f12-0efa072053af.png">

<img width="1134" alt="image" src="https://user-images.githubusercontent.com/97883527/160839801-db0699fd-6d78-4e6e-ab20-5409a4406caa.png">

amerkadicE commented on ui/pages/permissions-connect/redirect/permissions-redirect.component.js:
```diff
@@ -21,8 +21,9 @@ export default function PermissionsRedirect({ subjectMetadata }) {
+            <img src="./images/broken-line.svg" alt="broken-line" />
           </div>
           <SiteIcon
             icon="/images/logo/metamask-fox.svg"
```
Fixed

amerkadicE replied:
Fixed

amerkadicE replied:
Hi @georgewrmarshall , I just checked develop branch. Looks like we have different behaviour on the storybook and the MetaMask application. There is no broken line on the MetaMask application
Here is the screenshot from started application using develop branch
![image](https://user-images.githubusercontent.com/97883527/160849092-45fb6107-1a67-4d6b-aeb1-85b6bb4cce5f.png)

Here is screenshot form the storybook
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/97883527/160849992-1684ed62-e1af-4379-b19a-48b4fb8ab89a.png">

georgewrmarshall replied:
Fixed in updates PR merged into this one

brad-decker commented on app/images/icons/question-mark.svg:
```diff
@@ -0,0 +1,4 @@
+<svg width="80" height="80" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg">
```
These SVG files should be deleted if they aren't being used anymore :D

georgewrmarshall committed:
Icon and color consistency updates and to ensure theme (#14255)

compatibility

georgewrmarshall commented:
@amerkadicE merge my updates.

georgewrmarshall reviewed (commented):
UI LGTM! I think there is one level of <Box /> that can be removed which might have been added back with my updates so apologies but apart from that it's 💯 🔥

brad-decker commented:
> UI LGTM! I think there is one level of that can be removed which might have been added back with my updates so apologies but apart from that it's 100 fire

@georgewrmarshall  can you specify what needs to be removed? I'm not sure what is meant by this comment

Commit:
Remove one level of box element

Use question mark from fontawesome

Change question mark icon

Remove unkown picture and fix check icons

amerkadicE commented:
Hi @brad-decker Here is the design when using question mark from the font awesome. It looks slightly different. Does this look fine for you?
<img width="765" alt="image" src="https://user-images.githubusercontent.com/97883527/161256664-74a85412-8f8b-498e-8e88-5b97d95c2560.png">
<img width="765" alt="image" src="https://user-images.githubusercontent.com/97883527/161257238-7bbaeb98-aaf0-4e62-aae1-8c3f0fb08a4b.png">

brad-decker commented:
Hmm, No, i think its too different. At this point i'd advocate for adding the svg back -- we could probably tweak the styles of the fonticon to be smaller, and then use css to recreate the ring around the inner circle. If we could do that then the fonticon would be acceptable. However, right now the svg is dark mode compatible and is ready to go so lets go with that. Sorry!

georgewrmarshall commented:
@amerkadicE not sure how the font-awesome icon is implemented but you could try something like this

```
// scss
.network__unknown-icon {
  color: var(--color-icon-muted);
  border-radius: 50%;
  border: 1px solid var(--color-border-muted);
  background-color: var(--color-background-alternative);
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 64px;
  width: 64px;
}

// jsx
<div className="network__unknown-icon">
    <i className="fa fa-question fa-2x" />
</div>
```

Will render

<img width="151" alt="Screen Shot 2022-04-01 at 8 56 21 AM" src="https://user-images.githubusercontent.com/8112138/161299345-ed03ba9d-7298-405d-a23b-5b56150453df.png">

brad-decker reviewed (changes requested):
One other request: Could you add a  readme to ui/pages/confirmation/components that describes that the components in this folder are too specific to be used outside of the confirmation page and that they are not designed to be re-used. If any component in that folder should become eligible for reuse it should be elevated to the "global" components folder.

georgewrmarshall commented on ui/pages/permissions-connect/redirect/permission-redirect.stories.js:
```diff
@@ -1,21 +0,0 @@
-import React from 'react';
-import PermissionsRedirect from './permissions-redirect.component';
```
This was a duplicate file I believe

brad-decker replied:
But are teh changes to the permissions redirect page relevant? is it just to get the two components more inline so they can be extracted into a reusable component?

georgewrmarshall replied:
Yes, it is to make both of the components more consistent with each other. I will extract both components into a reusable component in another ticket.

amerkadicE commented on shared/constants/network.js:
```diff
@@ -68,6 +68,7 @@ export const TEST_ETH_TOKEN_IMAGE_URL = './images/black-eth-logo.svg';
 export const BNB_TOKEN_IMAGE_URL = './images/bnb.png';
 export const MATIC_TOKEN_IMAGE_URL = './images/matic-token.png';
 export const AVAX_TOKEN_IMAGE_URL = './images/avax-token.png';
+export const UNKNOWN_TOKEN_IMAGE_URL = './images/icons/question-mark.svg';
```
Done

danjm reviewed (approved):
LGTM

georgewrmarshall reviewed (approved):
LGTM! Nice work! 💯
